File: tennis-intelligence-platform/api/schemas/model_comparison.py
"""
schemas/model_comparison.py — Pydantic response model for the Model Comparison page.
Field names match tennis_intel.serving.model_comparison_service's own JSON export
shape exactly.
"""

from __future__ import annotations

from pydantic import BaseModel


class EngineStats(BaseModel):
    display_name: str
    n_points: int
    log_loss: float
    brier: float
    ece: float


class ModelComparisonResponse(BaseModel):
    n_matches: int
    n_points: int
    holdout_year: int
    is_full_holdout: bool
    engines: dict[str, EngineStats]

File: tennis-intelligence-platform-web/src/types/modelComparison.ts
// types/modelComparison.ts — matches api/schemas/model_comparison.py exactly.

export interface EngineStats {
  display_name: string;
  n_points: number;
  log_loss: number;
  brier: number;
  ece: number;
}

export interface ModelComparisonResponse {
  n_matches: number;
  n_points: number;
  holdout_year: number;
  is_full_holdout: boolean;
  engines: Record<string, EngineStats>;
}
